namerd/storage/consul/src/main/scala/io/buoyant/namerd/storage/consul/ConsulDtabStore.scala

```diff
@@ -29,7 +29,13 @@ class ConsulDtabStore(
   }
 
   override val list: Activity[Set[Ns]] = {
-    def namespace(key: String): Ns = key.stripPrefix("/").stripSuffix("/").substring(root.show.length)
+    def namespace(key: String): Option[Ns] =
+      Some(s"/$key".stripPrefix(s"${root.show}/"))
```

Comment:
This might benefit from an example to help see what's going on here.

If `key` is `/namerd/dtabs/foo` and `root` is `/namerd/dtabs`, then doesn't `s"/$key"` start with a double slash?

Reply:
I tried to illustrate that by an example in tests but probably that's to subtle.

Consul KV always returns paths without a leading `/` so we never going to get a double slash here. Should I add a comment explaining that?

Another way to achieve same behavior would be:
```scala
def namespace(key: String): Option[Ns] =
  Some(s"/$key")  // $key never has a leading slash
    .filterNot(_.endsWith("/")) // filter out dirs
    .map(_.stripPrefix(s"${root.show}/")) // strip $root
    .filter(namespaceIsValid)
```